pkg/controller/baremetalasset/baremetalasset_controller.go

```diff
@@ -295,13 +303,103 @@ func (r *ReconcileBareMetalAsset) Reconcile(request reconcile.Request) (reconcil
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileBareMetalAsset) checkHiveSyncSetInstance(bma *midasv1alpha1.BareMetalAsset, reqLogger logr.Logger) error {
+	found := &hivev1.SyncSetInstanceList{}
+	err := r.client.List(context.TODO(), found, client.MatchingLabels{hiveconstants.SyncSetNameLabel: bma.Name})
+
+	if err != nil {
+		reqLogger.Error(err, "Problem getting Hive SyncSetInstanceList")
+		return err
+	}
+	switch len(found.Items) {
+	case 0:
+		errorString := fmt.Sprintf("No SyncSetInstances with label name %v and label value %v found.", hiveconstants.SyncSetNameLabel, bma.Name)
+		conditionsv1.SetStatusCondition(&bma.Status.Conditions, conditionsv1.Condition{
+			Type:    midasv1alpha1.ConditionAssetSyncCompleted,
+			Status:  corev1.ConditionFalse,
+			Reason:  "SyncSetInstanceNotFound",
+			Message: errorString,
```

Comment:
```suggestion
			Message: err.Error(),
```
https://golang.org/src/builtin/builtin.go?s=11196:11236#L250